### ISSUE TITLE
Show Quicklinks in widget panel

### ISSUE THREAD

**Issue by mattiagalati** [kevinwucodes/gmail-quick-links#15]
The old and deprecated Quick Links widget provided by Gmail Labs was placed in a dedicated panel and activated by clicking on the three-dot icon on the bottom of the left panel.
This was very useful if one had a large amount of labels, since the widget panel is indipendent from the label tree.
Unfortunately, your Quick Links widget show at the bottom of label tree so the tree should be manually adjusted in height in order to make your widget always visible.

**Comment by kevinwucodes** [comment]
I'll do some research on this.  I think the Ellipsis menu comes from Gmail, so I would likely need to recreate that menu if the widget panel isn't enabled.

**Comment by hotmachiato** [comment]
I have a large label tree structure too and I always have to navigate on the pane above the fold to expose the Quick Links Widget. I found the [Gmail label column resizer](https://chrome.google.com/webstore/detail/gmail-label-column-resize/nopeaffbfnmebbnlpbfjemkfeojmlpoj/related?hl=en]) extension helps a bit in this respect, but it doesn't do exactly what I want

**Comment by fastcat** [comment]
Poking in my gmail view, the bar at the bottom with the various icons is a div with `class="J-KU-Jg J-KU-Jg-Zc aj5"`.  Within that there are the gmail-core items, but then there's another currently non-visible div with `class="J-KU"` and various other attrs labeling this as "Gadgets".  Within it is a single empty div with `class="aj4"`.

If I enable the Google Calendar lab, so that there's a Gadget to show, then the outer div gets an additional class `J-KU-KO` added and becomes visible, along with the inner `aj4` div which is the "..." button.

The Google Calendar widget is:
* `div class="T0 pp saH2Ef aj3"`
  * `div class="ApVoH"`
    * `div class="r"` -- this one contains the widget title and little "-" button to collapse it
    * `div class="pt"` -- this one contains the main calendar widget

There's a bunch of `id="..."` attrs on these.  It looks like additional widgets would be placed before/after the Calendar widget inside the `ApVoH` div.

Does this help?

**Comment by kevinwucodes** [comment]
@fastcat yes, this does!  I took another look and I think this is now doable

**Comment by kevinwucodes** [comment]
This should hit Chrome and Firefox shortly

**Comment by fastcat** [comment]
Whee :)

Slightly unfortunate it can't trigger the widgets panel to be visible on its own ... Turning on the Calendar widget just to relocate this widget is a little counter-intuitive ;)

**Comment by kevinwucodes** [comment]
I'll take another look

**Comment by fastcat** [comment]
This is yet more relevant in the new GMail UI, IMO -- positioning the inserted quick links panel above the Inbox and whatnot is frustratingly disruptive to the UI layout, at least to my eye.
